conjur/init_logic.py

```diff
@@ -0,0 +1,120 @@
+            # certificates and not use them to validate yet
+            ctx.check_hostname = False
+            ctx.verify_mode = False
+
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock = SSL.Connection(context=ctx, socket=sock)
+            sock.connect((hostname, port))
+            sock.do_handshake()
+        except Exception:
+            raise Exception(f"Unable to retrieve certificate from {hostname}:{port}")
+
+        logging.info("TLS Connection established. " \
+                     "Fetching certificate from server...")
+
+        return sock
+
+    # pylint: disable=unused-variable
+    @staticmethod
+    def get_certificate(hostname, port, sock, dump_certificate_func):
+        """
+        Method for connecting to Conjur to fetch the certificate chain
+        """
+        try:
+            chain = sock.get_peer_cert_chain()
+            cert = chain[0]
+            fingerprint = cert.digest("sha1").decode("utf-8")
+            # Get the peer chair and format to make it easier to later write to a file
+            readable_certificate=''
+            for (idx, cert) in enumerate(chain):
+                readable_certificate+=str(dump_certificate_func(FILETYPE_PEM, cert),  "utf-8")
+            logging.info("Certificates were fetched successfully")
+        except Exception:
+            raise Exception(f"Unable to retrieve certificate from {hostname}:{port}")
+
+        return fingerprint, readable_certificate
+
+    def write_certificate_to_file(self, fetched_certificate, cert_file_path, force_overwrite_flag):
+        """
+        Method for writing certificate details to a file on the user's machine
+        """
+        if fetched_certificate is not None:
+            self._check_file_exists(cert_file_path, force_overwrite_flag)
+            with open(cert_file_path, 'w') as config_fp:
+                config_fp.write(fetched_certificate)
+
+            sys.stdout.write(f"Wrote certificate to {cert_file_path}\n")
+
+    def write_conjurrc(self, conjurrc_file_path, input_obj, force_overwrite_flag):
+        """
+        Method for writing the conjurrc configuration
+        details needed to create a connection to Conjur
+        """
+        self._check_file_exists(conjurrc_file_path, force_overwrite_flag)
+
+        with open(conjurrc_file_path, 'w') as config_fp:
+            _pretty_print_object = {}
+
+            # Ensure no None fields are saved in conjurrc
+            for attr,value in input_obj.__dict__.items():
+                if value is not None:
+                    _pretty_print_object[str(attr)]=value
+
+            config_fp.write("---\n")
+            yaml.dump(_pretty_print_object, config_fp)
+
+        sys.stdout.write(f"Wrote configuration to {conjurrc_file_path}\n")
+
+    @staticmethod
+    def _check_file_exists(config_file, force_overwrite_flag):
+        if Path(config_file).is_file() and force_overwrite_flag is False:
```

Comment:
`os.path.exists` & `os.path.isfile` are better IMO

Reply:
updated to os.path.isfile. Easier for unit testing anyways 😉